Handle rejected promises from `router.param` (#92)
pillarjs/router: README.md

This function positions the middleware in the same stack as `.use`.

> If a `Promise` object is returned from the `param_middleware` function, the router

Reviewer: ```suggestion
If a `Promise` is returned from the `param_middleware` function, the router
```

> will attach an `onRejected` callback using `.then`. If the promise is rejected,

Reviewer: ```suggestion
will attach an `onRejected` callback using `.then(null, onRejected)`. If the promise is rejected,
```

I only think this could use more explicit call out because there is no commonly agreed upon wording I have seen other places.  Having the full explicit call might help make it more clear.

> `next` will be called with the rejected value, or an error if the value is falsy.

Reviewer: ```suggestion
`next` will be called with the rejected value as the error, or `new Error...` if the value is falsy.
```
